pkg/echo/proxy.go

```diff
@@ -13,17 +13,24 @@ import (
 // Proxy is a proxy type for logrus.Logger
 type Proxy struct {
 	dest *slog.Logger
+	ctx  context.Context
 }
 
 var proxy *Proxy
 
-// NewProxyFor creates a new Proxy for a particular logger
-func NewProxyFor(logger *slog.Logger) *Proxy {
+// NewProxyContextFor creates a new Proxy for a particular logger with a particular context.
+func NewProxyContextFor(logger *slog.Logger, ctx context.Context) *Proxy {
```

Comment:
The name seems to indicate this returns a new context, not a new proxy. Maybe `NewProxyWithContextFor`. I don't care very much it's just a common idiom to call things `NewT` :)